src/sdk/namespace_cache.js

```diff
@@ -357,8 +362,9 @@ class NamespaceCache {
             hub_read_stream.pipe(range_stream);
         }
 
-        // Object or part will only be uploaded to cache if size is not too big
-        if (hub_read_size <= params.bucket_free_space_bytes) {
+        // Object or part will only be uploaded to cache if size is not too big and
+        // the preconditions (if-match header etc.) are not set.
+        if (hub_read_size <= params.bucket_free_space_bytes && _.isUndefined(params.md_conditions)) {
```

Comment:
simplify check:
```suggestion
        if (hub_read_size <= params.bucket_free_space_bytes && params.md_conditions) {
```

Reply:
But actually why do preconditions should prevent us from putting to cache at all?